src/zenml/cli/stack.py

```diff
@@ -174,6 +174,200 @@ def register_stack(
+    "-x",
+    "--secrets_manager",
+    "secrets_manager_name",
+    help="Name of the new secrets manager for this stack.",
+    type=str,
+    required=False,
+)
+def update_stack(
+    stack_name: str,
+    metadata_store_name: Optional[str] = None,
+    artifact_store_name: Optional[str] = None,
+    orchestrator_name: Optional[str] = None,
+    container_registry_name: Optional[str] = None,
+    step_operator_name: Optional[str] = None,
+    secrets_manager_name: Optional[str] = None,
+) -> None:
+    """Update a stack."""
+    # TODO: [LOW] Implement a way to rename the stack
+    with console.status(f"Updating stack `{stack_name}`...\n"):
+        repo = Repository()
+        try:
+            current_stack = repo.get_stack(stack_name)
+        except KeyError:
+            cli_utils.error(
+                f"Stack `{stack_name}` cannot be updated as it does not exist.",
+            )
+        stack_components = current_stack.components
+
+        if metadata_store_name:
+            stack_components[
+                StackComponentType.METADATA_STORE
+            ] = repo.get_stack_component(
+                StackComponentType.METADATA_STORE, name=metadata_store_name
+            )
+
+        if artifact_store_name:
+            stack_components[
+                StackComponentType.ARTIFACT_STORE
+            ] = repo.get_stack_component(
+                StackComponentType.ARTIFACT_STORE, name=artifact_store_name
+            )
+
+        if orchestrator_name:
+            stack_components[
+                StackComponentType.ORCHESTRATOR
+            ] = repo.get_stack_component(
+                StackComponentType.ORCHESTRATOR, name=orchestrator_name
+            )
+
+        if container_registry_name:
+            stack_components[
+                StackComponentType.CONTAINER_REGISTRY
+            ] = repo.get_stack_component(
+                StackComponentType.CONTAINER_REGISTRY,
+                name=container_registry_name,
+            )
+
+        if step_operator_name:
+            stack_components[
+                StackComponentType.STEP_OPERATOR
+            ] = repo.get_stack_component(
+                StackComponentType.STEP_OPERATOR,
+                name=step_operator_name,
+            )
+
+        if secrets_manager_name:
+            stack_components[
+                StackComponentType.SECRETS_MANAGER
+            ] = repo.get_stack_component(
+                StackComponentType.SECRETS_MANAGER,
+                name=secrets_manager_name,
+            )
+
+        stack_ = Stack.from_components(
+            name=stack_name, components=stack_components
+        )
+        repo.update_stack(stack_)
+        cli_utils.declare(f"Stack `{stack_name}` successfully updated!")
+
+
+@stack.command(
+    "remove-component", context_settings=dict(ignore_unknown_options=True)
+)
+@click.argument("stack_name", type=str, required=True)
+@click.option(
+    "-c",
+    "--container_registry",
+    "container_registry_name",
```

Comment:
any reason to require the name? since there can only be one component of any given type in a stack, it should be obvious which one to remove when the type is known.

If I may offer a diverging suggestion how to handle the removing: simply add an additional keyword-option to the update stack command:
```
click.option(
    "--remove-component",
    type=click.choice(["container_registry", "step_operator","secrets_manager"]),
    multiple=True,
)
```
(untested,so the code may not quite run as written, but I think you get the gist)

Reply:
That's a good idea to ensure the names don't need to be written. I'll update this in the latest version somehow, though I might not stick exactly to your suggestion.